### PR TITLE
*: add a variable tidb_slow_log_threshold to set the slow log threshold dynamically

### PR DESCRIPTION
**Pull request by jackysp**
### What problem does this PR solve? 
Changing the slow log threshold in TiDB needs to reboot the server to affect.

### What is changed and how it works?
Add a variable "tidb_slow_log_threshold" to set the threshold by SQL.

### Check List 

Tests 

 - Unit test

Code changes

 - Has exported function/method change

 - Need to cherry-pick to the release branch

PTAL @tiancaiamao @lysu

### CODE REVIEW

File: config/config.go [hunk 1]
```diff
@@ -88,7 +88,7 @@ type Log struct {
 	File logutil.FileLogConfig `toml:"file" json:"file"`
 
 	SlowQueryFile      string `toml:"slow-query-file" json:"slow-query-file"`
-	SlowThreshold      uint   `toml:"slow-threshold" json:"slow-threshold"`
+	SlowThreshold      uint64 `toml:"slow-threshold" json:"slow-threshold"`
```
[the remark this answers is not in the document]
Comment (jackysp) [reply config/config.go]: Reduce some cast and easy to use the atomic package e.g. `atomic.StoreUint64`.

File: sessionctx/variable/varsutil.go [hunk 2]
```diff
@@ -85,6 +85,8 @@ func GetSessionOnlySysVars(s *SessionVars, key string) (string, bool, error) {
 		return string(j), true, nil
 	case TiDBForcePriority:
 		return mysql.Priority2Str[mysql.PriorityEnum(atomic.LoadInt32(&ForcePriority))], true, nil
+	case TiDBSlowLogThreshold:
+		return strconv.FormatUint(atomic.LoadUint64(&config.GetGlobalConfig().Log.SlowThreshold), 10), true, nil
```
Comment (lysu) [inline sessionctx/variable/varsutil.go]: IMHO, `adapator.go:351` also need change to `atomic.LoadUnit64` to keep visibility guarantee?

Comment (lysu) [reply sessionctx/variable/varsutil.go]: also better attach val to a session variable to reduce high rate atomic operation.

**Comment by morgo** [comment]
LGTM.  I like that it is in `ms`, and the default is 300.

**Comment by jackysp** [comment]
PTAL @lysu

**Comment by jackysp** [comment]
Won't support this until we support https://dev.mysql.com/doc/refman/8.0/en/persisted-system-variables.html.

**Pull request merged**